core/types/subtyping.cc

```diff
@@ -1032,6 +1032,9 @@ bool isSubTypeUnderConstraintSingle(Context ctx, TypeConstraint &constr, const T
                 int i = 0;
                 while (indexes[j] != a1->klass.data(ctx)->typeMembers()[i]) {
                     i++;
+                    if (i >= a1->klass.data(ctx)->typeMembers().size()) {
```

Comment:
Could we leave a comment on why this is necessary in code?

Reply:
Also, why does this _stop_ iteration over `j` entirely? I think we may need to validate other `j`s